src/analysis/ast_visitor.rs

```diff
@@ -0,0 +1,985 @@
+                                tuple_map
+                            });
+                            self.traverse_map_set(expr, name, &key, &value)
+                        }
+                        InsertEntry => {
+                            let name = args[0].match_atom().unwrap();
+                            let key = args[1].match_tuple().unwrap_or_else(|| {
+                                let mut tuple_map = HashMap::new();
+                                tuple_map.insert(None, &args[1]);
+                                tuple_map
+                            });
+                            let value = args[2].match_tuple().unwrap_or_else(|| {
+                                let mut tuple_map = HashMap::new();
+                                tuple_map.insert(None, &args[2]);
+                                tuple_map
+                            });
+                            self.traverse_map_insert(expr, name, &key, &value)
+                        }
+                        DeleteEntry => {
+                            let name = args[0].match_atom().unwrap();
+                            let key = args[1].match_tuple().unwrap_or_else(|| {
+                                let mut tuple_map = HashMap::new();
+                                tuple_map.insert(None, &args[1]);
+                                tuple_map
+                            });
+                            self.traverse_map_delete(expr, name, &key)
+                        }
+                        // TupleCons => self.traverse_tuple(&args),
+                        // TupleGet => {
+                        //     self.traverse_get(args[0].match_atom().unwrap(), &args[1], args.get(2))
+                        // }
+                        // Begin => self.traverse_begin(&args),
+                        // Hash160 | Sha256 | Sha512 | Sha512Trunc256 | Keccak256 => {
+                        //     traverse_hash(native_function, &args[0])
+                        // }
+                        // Print => self.traverse_print(&args[0]),
+                        // ContractCall => self.traverse_contract_call(
+                        //     args[0].match_atom().unwrap(),
+                        //     args[1].match_atom().unwrap(),
+                        //     &args[2..],
+                        // ),
+                        // AsContract => self.traverse_as_contract(&args[0]),
+                        // AtBlock => self.traverse_at_block(&args[0], &args[1]),
+                        // GetBlockInfo => {
+                        //     self.traverse_get_block_info(args[0].match_atom().unwrap(), &args[1])
+                        // }
+                        // ConsError => self.traverse_err(&args[0]),
+                        // ConsOkay => self.traverse_ok(&args[0]),
+                        // ConsSome => self.traverse_some(&args[0]),
+                        // DefaultTo => self.traverse_default_to(&args[0], &args[1]),
```

Comment:
Should we do something with this cases?

Reply:
Oh yes. All of the cases that weren't needed for the check-checker didn't get implemented yet. I'll add these in.

Reply:
Added traversers for all in 89c7611